YACS-RCOS/yacs.n
# Fast api conversion

**Issue**

Converted API from Flask to FastAPI and wrote test cases for all endpoints

> closes #522 

**Database Changes/Migrations**

No database changes

**Test Modifications**

Added unit tests for all api endpoints, new CI workflow step for API Tests.
Tests located in src/api/tests

**Test Procedure**

Checkout this branch and run

> docker-compose -f src/api/tests/docker-compose.yml up --build --abort-on-container-exit

This will execute all tests.
You can also run the development docker containers and go to localhost:5000/docs or localhost:5000/redoc to see all documentation for the new API

**Photos**

<img width="1273" alt="Screen Shot 2022-04-02 at 1 20 22 AM" src="https://user-images.githubusercontent.com/12472949/161367710-0c344397-267b-4f83-aa8f-cf131a10accc.png">
